src/v/kafka/server/replicated_partition.cc

```diff
@@ -314,11 +314,19 @@ ss::future<error_code> replicated_partition::validate_fetch_offset(
      * receive data.
      */
 
-    // calculate log end in kafka offset domain
-    const auto log_end = model::next_offset(
-      _translator->from_log_offset(_partition->dirty_offset()));
+    // Calculate log end in kafka offset domain
+    std::optional<model::offset> log_end;
+    if (_partition->dirty_offset() >= _partition->start_offset()) {
+        // Translate the raft dirty offset to find the kafka dirty offset.  This
+        // is conditional on dirty offset being ahead of start offset, because
+        // if it isn't, then the log is empty and we do not need to check for
+        // the case of a fetch between hwm and high watermark.  (Issue #7758)
```

Comment:
> between hwm and high watermark.

But hwm is high watermark, right? so probably high watermark -> dirty offset here.

Reply:
oops, fixed the typo.